### DISCUSSION TITLE
Livewire v3 - wire:loading.delay issue with file attachments

### DISCUSSION THREAD

**Discussion by rafalpietrzakio** [livewire/livewire#7891] [7. Bugs (archived)]
### Livewire version

v3.4.4

### Laravel version

10.43

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

1. Create component with file attachments
2. Add spinner component implementing wire:loading.delay.longer
3. Add files ($this->validate is called in updated lifecycle) - one or more files

In forms with file attachments, my wire:loading spinner gets 'stuck'; appears with delay (after the upload is finished) and does not disappear after it's done.

This happens **only** if I am using wire:loading.delay (or wire:loading.delay.longer), no issue with wire:loading without delay. It isn't a solution though as the spinner appears on every input now as each field is validated on updated.

Spinner appears (after delay - correctly), then disappears and around half second later appears again indefinitely or just appears with delay and does not disappear at all.

Open the form, select and choose files. *No throttling is necessary*

### Repository or Wirebox URL

https://github.com/rafalpietrzakio/livewire3-upload-wire-loading-issue

### Additional information

Reproduced on freshly installed laravel with livewire. My component uses tailwindcss classes by default, I've skipped tailwindcss for the purpose of reproduction and replaced with inline styles - although issue still appears

Reproduced repository has two 
/ - it's form with delayed spinner (issue appears)
/without-delay - it's form with no delay, spinner appears while the files are uploaded (instantly after files are selected) and correctly disappears when it's done

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@rafalpietrzakio Can you update to livewire v3.4.4 and see if the issue still happens? I know something like this got fixed not too long ago but I'm not sure what version.

**Comment by rafalpietrzakio** [comment]
Hi! Sorry, my bad in the description. It's the newest version of Livewire (3.4.4). Just updated the main thread. Unfortunately the issue appears here. I just upgraded from v2 to v3 recently, it worked entirely fine on v2, right now running with the newest v3 and trying to make some workarounds.

**Reply by joshhanley** [reply]
@rafalpietrzakio I had some issues with file upload loading too, which I fixed in PR #7897. It hasn't be released yet, but could you pull the main branch and test it to see if it fixes your issue? Thanks!

**Comment by rafalpietrzakio** [comment]
Hi! Thanks. I've just checked on that but unfortunately this doesn't fix it. The issue still appears. When delayed it now appears immediately with no delay BUT does not disappear at all. When not delayed it works properly - shows immediately and disappears on finished loading.  

You can check on the issue in this repo:
https://github.com/rafalpietrzakio/livewire3-upload-wire-loading-issue


route with loading.delay - 127.0.0.1:8000
route without loading.delay - 127.0.0.1:8000/no-delay
